gdx-video-teavm/src/com/badlogic/gdx/video/VideoPlayerTea.java

```diff
@@ -0,0 +1,179 @@
+/*******************************************************************************
+ * Copyright 2025 See AUTHORS file.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.badlogic.gdx.video;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.files.FileHandle;
+import com.badlogic.gdx.graphics.GL20;
+import com.badlogic.gdx.graphics.Pixmap;
+import com.badlogic.gdx.graphics.Texture;
+import com.badlogic.gdx.utils.Null;
+import com.github.xpenatan.gdx.backends.teavm.TeaFileHandle;
+import com.github.xpenatan.gdx.backends.teavm.TeaGL20;
+import com.github.xpenatan.gdx.backends.teavm.dom.HTMLVideoElementWrapper;
+import org.teavm.jso.browser.Window;
+import org.teavm.jso.dom.html.HTMLDocument;
+import org.teavm.jso.dom.html.HTMLVideoElement;
+
+import java.io.FileNotFoundException;
+
+public class VideoPlayerTea extends AbstractVideoPlayer implements VideoPlayer {
+	private static final HTMLDocument document = Window.current().getDocument();
+	private FileHandle currentFile;
+	private final HTMLVideoElement v;
+	private Texture frame;
+	private int width, height;
+	private VideoSizeListener videoSizeListener;
+
+	public VideoPlayerTea () {
+		v = document.createElement("video").cast();
+	}
+
+	@Override
+	public boolean load (FileHandle file) throws FileNotFoundException {
+		currentFile = file;
+		if (v != null) {
+			v.setSrc(((TeaFileHandle)file).getAssetUrl());
+			v.load();
+			return true;
+		}
+		return false;
+	}
+
+	@Override
+	public void play () {
+		if (v != null) {
+			v.play();
+		}
+	}
+
+	@Override
+	public boolean update () {
+		if (v != null) {
+			if (v.getVideoHeight() != height || v.getVideoWidth() != width) {
+				height = v.getVideoHeight();
+				width = v.getVideoWidth();
+
+				if (videoSizeListener != null) videoSizeListener.onVideoSize(width, height);
+			}
+			if ((!v.isPaused() || v.getCurrentTime() == 0) && isBuffered() && width * height > 0) {
+				if (frame != null && (frame.getWidth() != width || frame.getHeight() != height)) {
+					frame.dispose();
+					frame = null;
+				}
+				if (frame == null) {
+					frame = new Texture(width, height, Pixmap.Format.RGB888);
+					frame.setFilter(minFilter, magFilter);
+				}
+				frame.bind();
+				((TeaGL20)Gdx.gl).gl.texImage2D(GL20.GL_TEXTURE_2D, 0, GL20.GL_RGB, GL20.GL_RGB, GL20.GL_UNSIGNED_BYTE,
+					(HTMLVideoElementWrapper)v.cast());
```

Comment:
Hi, I think this will be an issue when updating to 1.2.0.  I Removed the WebGLRenderingContext wrapper and i'll later remove other wrappers also. It seems that the teavm version dont have texImage2D with HTMLVideoElement parameter.

What you can do so updating to later version wont give issues it creating a interface that have this parameter. Ex:

public interface WebGLRenderingContextExt extends JSObject {
   void texImage2D(int target, int level, int internalformat, int format, int type, HTMLVideoElement video);
}

To use this interface you can just cast the interface from ((TeaGL20)Gdx.gl).gl to WebGLRenderingContextExt.